python/mlc_chat/compiler/quantization/group_quantization.py

```diff
@@ -102,15 +149,27 @@ def _dequantize(
         return dequantized_weight
 
     def quantize_weight(self, weight: NDArray) -> List[NDArray]:
-        """Quantize weight with group quantization"""
+        """
+        Quantize weight with group quantization
+
+        Parameters
+        ----------
+        weight : NDArray
+            The original weight.
+
+        Returns
+        ------
+        ret: List[NDArray]
+            The list of group quantized weights.
+        """
         assert weight.dtype == self.model_dtype
         assert len(weight.shape) == 2
-        bb = relax.BlockBuilder()
+        bb = relax.BlockBuilder()  # pylint: disable=invalid-name
```

Comment:
A couple of things to improve in this code snippet:
- Don't hard code cuda here, because dlight is supposed to work on all GPUs, not only CUDA. The target could be detected via `weight.device` if not provided
- Save compilation time by caching the compiled IRModule in `GroupQuantize` if the same shape/dtype appears before
- As @zxybazh identified, but in an opposite direction, let's assume the input and output of this method resides already on the desired device, and let the logics outside to handle the rest

Reply:
Fixed as the third bullet.